### DISCUSSION TITLE
Physical buttons

### DISCUSSION THREAD

**Discussion by kristianolofsson** [Blackymas/NSPanel_HA_Blueprint#825] [General]
Can a script be triggered upon single click of any of the buttons? If not sure I can trigger by condition light is on.
But how to call a script or similar to change the page of the spaniel?

I would like to call the dimmer-slider upon single click so it works like a "normal" dimmer. Possible?

I know the button-hold solves this but do not really like that. Single klick is better.

**Comment by edwardtfn** [comment]
You can assign an existing script to a button so the script will run every time the button is pressed.

I'm not sure if I got your question.

**Comment by kristianolofsson** [comment]
Sure A script would help. Where to initiate the script and how to trigger a page change?

`service: esphome.office_workstation_panel_set_settings_entity
data:
  entity: '{"page": "home", "entity": "light.office_ceiling_lines"}' `

**Reply by edwardtfn** [reply]
You mean assign a script or a service call?

Anyways, there's no way to assign a service call directly to a button, so you have to create a script (Setting > Automations & Scenes then select tab Scripts) and add there your service call. Then you assign this new script to a button.

By the way, the service to open a specific page is like this:

```yaml 
service: esphome.xxxxx_send_command_printf
data:
  cmd: page buttonpage02
```

However, if you want to open one of the setting page, then is probably better to use the other service from your post above, as it will inform which entity should be shown.
